py/thermostat/configure_output_ch.py

```diff
@@ -0,0 +1,175 @@
+        default=0,
+        help="The channel input offset (%(default)s V)",
+    )
+    parser.add_argument(
+        "--y-min",
+        type=float,
+        default=-1,
+        help="The channel minimum output (%(default)s V)",
+    )
+    parser.add_argument(
+        "--y-max",
+        type=float,
+        default=1,
+        help="The channel maximum output (%(default)s V)",
+    )
+    parser.add_argument(
+        "--y-offset",
+        type=float,
+        default=0,
+        help="The channel output offset (%(default)s V)",
+    )
+    parser.add_argument(
+        "--input-weights",
+        type=float,
+        default=[1, 0, 0, 0, 0, 0, 0, 0],
+        help="Input channel weights (%(default)s V)",
+    )
+    parser.add_argument(
+        "--shutdown",
+        type=bool,
+        default=False,
+        help="Channel TEC shutdown (%(default)s V)",
+    )
+    parser.add_argument(
+        "--hold", type=bool, default=False, help="Channel IIR hold (%(default)s V)"
+    )
+    parser.add_argument(
+        "--voltage-limit",
+        type=float,
+        default=1,
+        help="Channel voltage limit (%(default)s V)",
+    )
+
+    # Next, add subparsers and their arguments.
+    subparsers = parser.add_subparsers(
+        help="Filter-specific design parameters", dest="filter_type", required=True
+    )
+
+    filters = get_filters()
+
+    for (filter_name, filt) in filters.items():
+        subparser = subparsers.add_parser(filter_name, help=filt.help)
+        for arg in filt.arguments:
+            subparser.add_argument(*arg.positionals, **arg.keywords)
+
+    args = parser.parse_args()
+
+    logging.basicConfig(
+        format="%(asctime)s [%(levelname)s] %(name)s: %(message)s",
+        level=logging.WARN - 10 * args.verbose,
+    )
+
+    # Calculate the IIR coefficients for the filter.
+    coefficients = filters[args.filter_type].coefficients(args)
+
+    # The feed-forward gain of the IIR filter is the summation
+    # of the "b" components of the filter.
+    forward_gain = sum(coefficients[:3])
+    if forward_gain == 0 and args.x_offset != 0:
+        logger.warning("Filter has no DC gain but x_offset is non-zero")
+
+    if args.no_discover:
+        prefix = args.prefix
+    else:
+        devices = asyncio.run(miniconf.discover(args.broker, args.prefix))
+        if not devices:
+            raise ValueError("No prefixes discovered.")
+        if len(devices) > 1:
+            raise ValueError(
+                f"Multiple prefixes discovered ({devices})."
+                "Please specify a more specific --prefix"
+            )
+        prefix = devices[0]
+        logger.info("Automatically using detected device prefix: %s", prefix)
+
+    async def configure():
+        logger.info("Connecting to broker")
+        interface = await miniconf.Miniconf.create(prefix, args.broker)
+
+        # Set the filter coefficients.
+        await interface.command(
+            f"output_channel/{args.channel}",
```

Comment:
Ah. Does that have to be a `MiniconfAtomic`? IMO having `iir` as `MiniconfAtomic` is sufficient. The rest doesn't need to be synchronized.

Reply:
https://github.com/quartiq/thermostat-eem/blob/d8aef7332822e2c04ced8d273dbbede92d5501c3/src/output_channel.rs#L8

Reply:
Ah I remember why I did it this way. Because the weights then become an individual path each.